app/javascript/listings/components/ListingTagsField.jsx

```diff
@@ -0,0 +1,121 @@
+import { h } from 'preact';
+import { useEffect, useMemo, useState } from 'preact/hooks';
+import PropTypes from 'prop-types';
+import { useTagsField } from '../../hooks/useTagsField';
+import { TagAutocompleteOption } from '../../article-form/components/TagAutocompleteOption';
+import { TagAutocompleteSelection } from '../../article-form/components/TagAutocompleteSelection';
+import { MultiSelectAutocomplete } from '@crayons';
+
+export const ListingTagsField = ({
+  defaultValue,
+  onInput,
+  categorySlug,
+  name,
+  onFocus,
+}) => {
+  const listingState = useMemo(
+    () => ({
+      additionalTags: {
+        jobs: [
+          'remote',
+          'remoteoptional',
+          'lgbtbenefits',
+          'greencard',
+          'senior',
+          'junior',
+          'intermediate',
+          '401k',
+          'fulltime',
+          'contract',
+          'temp',
+        ],
+        forhire: [
+          'remote',
+          'remoteoptional',
+          'lgbtbenefits',
+          'greencard',
+          'senior',
+          'junior',
+          'intermediate',
+          '401k',
+          'fulltime',
+          'contract',
+          'temp',
+        ],
+        forsale: ['laptop', 'desktopcomputer', 'new', 'used'],
+        events: ['conference', 'meetup'],
+        collabs: ['paid', 'temp'],
+      },
+    }),
+    [],
+  );
+
+  const { defaultSelections, topTags, fetchSuggestions, syncSelections } =
+    useTagsField({ defaultValue, onInput });
+  const [topTagsWithAdditional, setTopTagsWithAdditional] = useState([
+    ...topTags,
+  ]);
+
+  useEffect(() => {
+    if (listingState && listingState.additionalTags) {
+      const newTagsForSelectedCategory =
+        listingState.additionalTags[categorySlug] || [];
+      // Push in this way: { name: 'remote' }
+      const formattedNewTagsForSelectedCategory =
+        newTagsForSelectedCategory.map((name) => ({ name }));
+      setTopTagsWithAdditional([
+        ...topTags,
+        ...formattedNewTagsForSelectedCategory,
+      ]);
```

Comment:
Update - I spoke to our Product team, and in this location we _don't_ want to show top tags when the field is empty. We only want to show the category suggested tags (e.g. 'remote', etc). This will help users see the most relevant tags when creating a listing.

Are you happy to update with that mind? One thing to consider is that the `useTagsField` hook shouldn't make any network request for the top tags on this page, since we don't need them here.

(sorry this wasn't anticipated/defined up front!)